src/ui/layout/tutorial.md

```diff
@@ -244,219 +393,287 @@ class MyApp extends StatelessWidget {
-to that column. Align the columns along the main axis
-using `MainAxisAlignment.spaceEvenly` to arrange the
-free space evenly before, between, and after each column.
-Add the following code just below the
-`titleSection` declaration inside the `build()` method:
+Add the following code into the `ButtonSection` widget.
 
-<?code-excerpt "lib/main.dart (buttonSection)" title?>
+1. Call the helper method three times, once for each button.
+1. Pass the color, `Icon`, and text for that specific button.
+1. Align the columns along the main axis with the
+   `MainAxisAlignment.spaceEvenly` method.
+   This arranges the free space in equal amounts before, between,
+   and after each column.
+
+<?code-excerpt "lib/main.dart (ButtonSection)" title?>
 ```dart
-Color color = Theme.of(context).primaryColor;
-
-Widget buttonSection = Row(
-  mainAxisAlignment: MainAxisAlignment.spaceEvenly,
-  children: [
-    _buildButtonColumn(color, Icons.call, 'CALL'),
-    _buildButtonColumn(color, Icons.near_me, 'ROUTE'),
-    _buildButtonColumn(color, Icons.share, 'SHARE'),
-  ],
-);
-```
+class ButtonSection extends StatelessWidget {
+  const ButtonSection({super.key});
 
-Add the button section to the body:
+  @override
+  Widget build(BuildContext context) {
+    final Color color = Theme.of(context).primaryColor;
+    return SizedBox(
+      child: Row(
+        mainAxisAlignment: MainAxisAlignment.spaceEvenly,
+        children: [
+          BuildButtonColumn(
+            color: color,
+            icon: Icons.call,
+            label: 'CALL',
+          ),
+          BuildButtonColumn(
+            color: color,
+            icon: Icons.near_me,
+            label: 'ROUTE',
+          ),
+          BuildButtonColumn(
+            color: color,
+            icon: Icons.share,
+            label: 'SHARE',
+          ),
+        ],
+      ),
+    );
+  }
+}
 
-<?code-excerpt path-base="layout/lakes"?>
-<?code-excerpt "{step2,step3}/lib/main.dart" from="return MaterialApp" to="}"?>
-```diff
---- step2/lib/main.dart
-+++ step3/lib/main.dart
-@@ -48,3 +59,3 @@
-     return MaterialApp(
-       title: 'Flutter layout demo',
-       home: Scaffold(
-@@ -54,8 +65,9 @@
-         body: Column(
-           children: [
-             titleSection,
-+            buttonSection,
-           ],
-         ),
-       ),
-     );
-   }
+class BuildButtonColumn extends StatelessWidget {
+  const BuildButtonColumn({
+    super.key,
+    required this.color,
+    required this.icon,
+    required this.label,
+  });
+
+  final Color color;
+  final IconData icon;
+  final String label;
+
+  @override
+  Widget build(BuildContext context) {
+    return Column(
+// ···
+    );
+  }
+}
 ```
 
-## Step 4: Implement the text section
+### Update the app to display the button section
+
+Add the button section to the `children` array.
+
+    <?code-excerpt path-base="layout/lakes"?>
+    
+    <?code-excerpt "step{2,3}/lib/main.dart (addWidget)" title?>
```

Comment:
This snippet is failing to render: https://flutter-docs-prod--pr9936-revert-9927-revert-9893-fix-clit72es.web.app/ui/layout/tutorial#update-the-app-to-display-the-button-section

Perhaps because it's indented? Is there a particular reason these instructions and the snippet were intented here?

Reply:
Fixed.